common:
    lid_topic: "/livox/lidar"
    imu_topic: "/mavros/imu/data"
    time_offset_lidar_imu: 0.0
    drone_id: 1

multiuav:
    cluster_extraction_in_predict_region: true
    inten_threshold: 240
    min_high_inten_cluster_size: 4
    min_cluster_size: 7
    predict_region_radius: 0.5
    temp_predict_region_radius: 0.5
    traj_matching_start_thresh: 100.0
    ave_match_error_thresh: 0.2
    pos_num_in_traj: 300
    reset_tracker_thresh: 5.0
    valid_cluster_size_thresh: 0.5
    valid_cluster_dist_thresh: 0.2
    mutual_observe_noise: 0.001
    frame_num_in_sliding_window: 3
    actual_uav_num: 5
    degeneration_thresh: 10.0

imu_propagate:
    enable: true
    topic: "/lidar_slam/imu_propagate"
    filter_acc_en: false

preprocess:
    lidar_type: 1  # mid360
    feature_extract_en: false
    scan_line: 4
    blind: 0.2

calibration:
    cut_frame_en: false
    cut_frame_num: 1
    accumulate_frame_en: true
    original_frequency: 30 # hz
    gravity_align_enable: true
    gravity_align_time: 10 # seconds
    PX4_R_I: [0.0, 0.0, 0.0] # degree
    PX4_p_I: [0.0, 0.0, 0.0] # meter
    level_horizon: [0.0, 0.0, 0.0] #degree

mapping:
    imu_en: true
    acc_cov: 0.1
    gyr_cov: 0.1
    b_acc_cov: 0.0001
    b_gyr_cov: 0.0001
    det_range: 100.0
    LI_extrinsic_T: [ -0.019391, -0.000278,  0.080926]
    LI_extrinsic_R: [ 1, 0, 0, 0, 1, 0, 0, 0, 1]
    point_filter_num: 2
    max_iteration: 4
    filter_size_surf: 0.2
    filter_size_map: 0.3
    cube_side_length: 2000

publish:
    path_en: true
    scan_publish_en: true       # false: close all the point cloud output
    dense_publish_en: true       # false: low down the points number in a global-frame point clouds scan.
    scan_bodyframe_pub_en: false  # true: output the point cloud scans in IMU-body-frame

pcd_save:
    pcd_save_en: false
    pcd_resolution: 0.1
    interval: -1                 # how many LiDAR frames saved in each pcd file;
    # -1 : all frames will be saved in ONE pcd file, may lead to memory crash when having too many frames.
